lib/GenomeFileUtil/core/GenomeInterface.py

```diff
@@ -128,54 +173,70 @@ def get_one_genome(self, params):
-            workspace_id = self.dfu.ws_name_to_id(workspace)
-
-        save_params = {'id': workspace_id,
-                       'objects': [{'type': ws_datatype,
-                                    'data_json_file': data_path,
-                                    'name': name,
-                                    'meta': meta,
-                                    'hidden': hidden}]}
-        dfu_oi = self.ws_large_data.save_objects(save_params)[0]
-        returnVal = {'info': dfu_oi, 'warnings': data.get('warnings', [])}
-        return returnVal
+        workspace_id = params[_WSID]
+        inputs = params[_INPUTS]
+
+        objects = []
+        warnings = []
+
+        for input_params in inputs:
+
+            obj = {}
+
+            # retrive required params
+            name = input_params['name']
+            data = dict(input_params['data'])
+
+            # XXX there is no `workspace_datatype` param in the spec
+            # NOTE: This allows a user to specify any arbitrary workspace type which could cause,
+            # in the worst case, data corruption. It should be removed from the API
+            # (note it is not currently documented) so users cannot access it.
+            ws_datatype = input_params.get('workspace_datatype', "KBaseGenomes.Genome")
+            # XXX there is no `meta` param in the spec
+            meta = input_params.get('meta', {})
+
+            obj["type"] = ws_datatype
+            obj["name"] = name
+            obj["meta"] = meta
+
+            if "AnnotatedMetagenomeAssembly" not in ws_datatype:
+                if input_params.get('upgrade') or 'feature_counts' not in data:
+                    data = self._update_genome(data)
+
+            # check all handles point to shock nodes owned by calling user
+            self._own_handle(data, 'genbank_handle_ref')
+            self._own_handle(data, 'gff_handle_ref')
+            if "AnnotatedMetagenomeAssembly" not in ws_datatype and validate_genome:
+                self.check_dna_sequence_in_features(data)
+                data['warnings'] = self.validate_genome(data)
+
+            # sort data
+            data = GenomeUtils.sort_dict(data)
+            # dump genome to scratch for upload
+            data_path = os.path.join(self.scratch, name + ".json")
+            json.dump(data, open(data_path, 'w'))
+            if 'hidden' in input_params and str(input_params['hidden']).lower() in ('yes', 'true', 't', '1'):
+                hidden = 1
+            else:
+                hidden = 0
+
+            obj["data_json_file"] = data_path
+            obj["hidden"] = hidden
+
+            objects.append(obj)
+            warnings.append(data.get('warnings', []))
+
+        dfu_infos = self.ws_large_data.save_objects(
+            {'id': workspace_id, 'objects': objects}
+        )
+
+        output = [
+            {'info': dfu_oi, 'warnings': warning}
+            for dfu_oi, warning in zip(dfu_infos, warnings)
+        ]
+        return output
```

Comment:
This is returned directly by G2G, but doesn't match what's in the spec. Am I missing something?

Reply:
Which part does not match the spec?

Reply:
This is so long ago I don't remember what I was thinking, but 

> This is returned directly by G2G

appears to be wrong - I think I missed a code fold